File: go/pkg/models/album.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Album struct {
	ID        uuid.UUID `gorm:"type:uuid;primary_key;default:uuid_generate_v4()" json:"id"`
	CreatedAt time.Time `gorm:"not null" json:"createdAt"`
	UpdatedAt time.Time `gorm:"not null" json:"updatedAt"`
	UserID    uuid.UUID `gorm:"type:uuid;not null;index" json:"userId"`
	Name      string    `gorm:"type:text;not null" json:"name"`
}

type AlbumTag struct {
	AlbumID uuid.UUID `gorm:"type:uuid;primaryKey" json:"albumId"`
	Tag     string    `gorm:"type:text;primaryKey" json:"tag"`
}

File: dart/skunkworks/lib/models/album.dart
class Album {
  final String? id;
  final DateTime? createdAt;
  final DateTime? updatedAt;
  final String? userId;
  final String name;
  final List<String> tags;

  const Album({
    this.id,
    this.createdAt,
    this.updatedAt,
    this.userId,
    required this.name,
    this.tags = const [],
  });

  factory Album.fromJson(Map<String, dynamic> json) {
    return Album(
      id: json['id']?.toString(),
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'] as String)
          : null,
      updatedAt: json['updatedAt'] != null
          ? DateTime.tryParse(json['updatedAt'] as String)
          : null,
      userId: json['userId']?.toString(),
      name: json['name'] as String? ?? '',
      tags: json['tags'] != null
          ? (json['tags'] as List<dynamic>).map((t) => t.toString()).toList()
          : [],
    );
  }
}
